nexus/src/app/support_bundles.rs

```diff
@@ -0,0 +1,130 @@
+    ) -> ListResultVec<SupportBundle> {
+        self.db_datastore.support_bundle_list(&opctx, pagparams).await
+    }
+
+    pub async fn support_bundle_view(
+        &self,
+        opctx: &OpContext,
+        id: SupportBundleUuid,
+    ) -> LookupResult<SupportBundle> {
+        self.db_datastore.support_bundle_get(&opctx, id).await
+    }
+
+    pub async fn support_bundle_create(
+        &self,
+        opctx: &OpContext,
+        reason: &'static str,
+    ) -> CreateResult<SupportBundle> {
+        self.db_datastore.support_bundle_create(&opctx, reason, self.id).await
+    }
+
+    pub async fn support_bundle_download(
+        &self,
+        opctx: &OpContext,
+        id: SupportBundleUuid,
+        query: &SupportBundleGetQueryParams,
+        head: bool,
+        _range: Option<PotentialRange>,
+    ) -> Result<Response<Body>, Error> {
+        // Lookup the bundle, confirm it's accessible
+        let bundle = self.db_datastore.support_bundle_get(&opctx, id).await?;
+        if !matches!(bundle.state, SupportBundleState::Active) {
+            return Err(Error::invalid_request(
+                "Cannot download bundle in non-active state",
+            ));
+        }
+
+        // Lookup the sled holding the bundle, forward the request there
+        let sled_id = self
+            .db_datastore
+            .zpool_get_sled(&opctx, bundle.zpool_id.into())
+            .await?;
+        let client = self.sled_client(&sled_id).await?;
+
+        // TODO: Use "range"?
+
+        let response = if head {
+            client
+                .support_bundle_head(
+                    &ZpoolUuid::from(bundle.zpool_id),
+                    &DatasetUuid::from(bundle.dataset_id),
+                    &SupportBundleUuid::from(bundle.id),
+                    &query,
+                )
+                .await
+        } else {
+            client
+                .support_bundle_get(
+                    &ZpoolUuid::from(bundle.zpool_id),
+                    &DatasetUuid::from(bundle.dataset_id),
+                    &SupportBundleUuid::from(bundle.id),
+                    &query,
+                )
+                .await
+        };
+
+        let response =
+            response.map_err(|err| Error::internal_error(&err.to_string()))?;
+
+        // The result from the sled agent a "ResponseValue<ByteStream>", but we
+        // need to coerce that type into a "Response<Body>" while preserving the
+        // status, headers, and body.
+        let mut builder = Response::builder().status(response.status());
+        let headers = builder.headers_mut().unwrap();
+        headers.extend(
+            response.headers().iter().map(|(k, v)| (k.clone(), v.clone())),
+        );
+        let body = http_body_util::StreamBody::new(
+            response
+                .into_inner_stream()
+                .map_ok(|b| hyper::body::Frame::data(b)),
+        );
+        Ok(builder.body(Body::wrap(body)).unwrap())
```

Comment:
My intent here is to act as much like a transparent proxy as I can -- find the sled that should hold the bundle, and send a request there.

If it's possible to do that without these types, I'd love that.